go/master/service.go

```diff
@@ -77,11 +77,13 @@ type taskEntry struct {
 	NumFailure int
 }
 
-type taskQueues struct {
-	Todo    []taskEntry
-	Pending map[int]taskEntry // map from task ID to task entry
-	Done    []taskEntry
-	Failed  []taskEntry
+type masterState struct {
+	Todo     []taskEntry
+	Pending  map[int]taskEntry // map from task ID to task entry
+	Done     []taskEntry
+	Failed   []taskEntry
+	CurPass  int
+	JobTasks []taskEntry
```

Comment:
JobTasks stays the same across one job, maybe no need to save it every time? Anyway, this doesn't' harm.